packages/types/config/module.d.ts

```diff
@@ -17,10 +19,35 @@ interface ExtendFunctionContext {
   loaders: NuxtOptionsLoaders
 }
 
-type ExtendFunction = (config: WebpackConfiguration, ctx: ExtendFunctionContext) => void
+export interface ModuleTemplateConfig {
+  src: string,
+  fileName?: string,
```

Comment:
- We have both `fileName?` and `filename?`
- Options can be types as `Record<string, any>`
- We also allow `[key: string]: any` additional properties for custom purposes